hooks/use-upload.ts

```diff
@@ -0,0 +1,68 @@
+import {useCallback, useState} from 'react'
+import {AxiosResponse} from 'axios'
+const axios = require('axios')
+
+export const useUpload = (params: {
+  fileFieldName?: string
+  onError?: (error) => void
+  onFinish?: () => void
+  onSuccess?: (response: AxiosResponse<{location: string}>) => void
+  onStart?: (CancelTokenSource) => void
+  setUploadProgress?: (progress: number) => void
+  urlOrPath: string
+}): {
+  isUploading: boolean
+  isUploadError: boolean
+  upload: (file: Blob) => Promise<AxiosResponse | undefined>
+} => {
+  const [isUploading, setStateIsUploading] = useState<boolean>(false)
+  const [isUploadError, setIsUploadError] = useState<any>(false)
+
+  const upload = useCallback(async (file: Blob) => {
+    const CancelToken = axios.CancelToken
+    const source = CancelToken.source()
+
+    const config = {
+      cancelToken: source.token,
+
+      onUploadProgress: progressEvent => {
+        if(typeof params.setUploadProgress === 'function') {
+          params.setUploadProgress(Math.round(progressEvent.loaded * 100 / progressEvent.total))
+        }
+      },
+    }
+
+    const data = new FormData()
+    data.append(params.fileFieldName || 'file', file[0])
```

Comment:
@pufflik I guess it shouldn't be an array:
```suggestion
    data.append(params.fileFieldName || 'file', file)
```

This `blob[0]` instead of `blob` must be a legacy:
https://github.com/ottofeller/classo/blob/f5b618a071a01873e77828082a4ab9c7a1d96469/frontend/components/teacher/dashboard/Profile/Form/index.tsx#L66

Because this part could potentially send an array of images:
https://github.com/ottofeller/classo/blob/f5b618a071a01873e77828082a4ab9c7a1d96469/frontend/components/teacher/dashboard/Profile/Form/index.tsx#L163

https://github.com/ottofeller/classo/blob/f5b618a071a01873e77828082a4ab9c7a1d96469/frontend/components/common/upload/Button/index.tsx#L20